test/flink_test.go

```diff
@@ -120,3 +152,244 @@ func (s *CLITestSuite) TestFlink_Autocomplete() {
+	if err := os.Setenv(timezoneEnvVar, "Europe/London"); err != nil {
+		return envVars, err
+	}
+	envVars = append(envVars, timezoneEnvVar)
+
+	return envVars, nil
+}
+
+func cleanupEnvVars(envVars []string) {
+	for _, envVar := range envVars {
+		os.Unsetenv(envVar)
+	}
+}
+
+func killCommandIfTimeoutExceeded(timeout time.Duration, cmd *exec.Cmd) {
+	if timeout == 0 {
+		timeout = 10 * time.Second
+	}
+	time.Sleep(timeout)
+	if err := cmd.Process.Kill(); err != nil {
+		fmt.Printf("failed to kill command %v/n", err)
+	}
+}
+
+func waitForFlinkShellToBeReady(scanner *bufio.Scanner, output *strings.Builder) {
+	for scanner.Scan() {
+		line := scanner.Text()
+		output.WriteString(line + "\n")
+		if strings.Contains(line, "[Ctrl-Q] Quit [Ctrl-S] Toggle Smart Completion") {
+			break
+		}
+	}
+}
+
+func executeCommands(stdin *os.File, commands []string, stdoutScanner *bufio.Scanner, output *strings.Builder) error {
+	// add exit command to ensure we always close the flink shell
+	commands = append(commands, "exit")
+	for _, command := range commands {
+		// Simulate the user entering a command and add a new line to flush the output buffer
+		_, err := stdin.WriteString(command + "\n")
+		if err != nil {
+			return err
+		}
+
+		for stdoutScanner.Scan() {
+			// Strip all terminal control sequences and skip empty lines
+			line := removeAnsiEscapeSequences(stdoutScanner.Text())
+			if line == "" {
+				continue
+			}
+
+			// Record the output
+			output.WriteString(line + "\n")
+
+			// Once we've seen our command be printed, we know we can continue to submit the statement.
+			// This effectively means we wait for the previous command to complete, since the new command will only
+			// be printed, if the previous one is finished.
+			if strings.Contains(line, fmt.Sprintf("> %s", command)) {
+				err := submitStatement(stdin)
+				if err != nil {
+					return err
+				}
+				break
+			}
+		}
+	}
+	return nil
+}
+
+func removeAnsiEscapeSequences(input string) string {
+	stripColors := regexp.MustCompile(`\x1b\[[0-9;]*[JKmsu]`)
+	stripBellCharacter := regexp.MustCompile(`\a`)
+	stripTitle := regexp.MustCompile(`\x1B]2;`)
+	stripSqlPrompt := regexp.MustCompile(`sql-prompt`)
+	stripErasePrefix := regexp.MustCompile(`> \x1b\[2D`)
+	stripCursorUp := regexp.MustCompile(`\x1b\[1A.*`)
+
+	input = stripColors.ReplaceAllString(input, "")
+	input = stripBellCharacter.ReplaceAllString(input, "")
+	input = stripTitle.ReplaceAllString(input, "")
+	input = stripSqlPrompt.ReplaceAllString(input, "")
+	input = stripErasePrefix.ReplaceAllString(input, "")
+	input = stripCursorUp.ReplaceAllString(input, "")
+
+	return strings.TrimSpace(input)
+}
+
+func submitStatement(stdin *os.File) error {
+	// We need to wait before and after writing a new line, so go-prompt receives the ENTER as an individual key press
+	time.Sleep(50 * time.Millisecond)
+	_, err := stdin.WriteString("\n")
+	time.Sleep(50 * time.Millisecond)
+	return err
```

Comment:
If there was an error, wouldn't we want to return it right away instead of waiting 50ms?

Reply:
should be resolved as well with the latest commit